src/sagemaker/estimator.py

```diff
@@ -3862,10 +3863,11 @@ def _distribution_configuration(self, distribution):
             elif isinstance(self.instance_type, str):
                 p5_enabled = "p5.48xlarge" in self.instance_type
             else:
-                raise ValueError(
-                    "Invalid object type for instance_type argument. Expected "
-                    f"{type(str)} or {type(ParameterString)} but got {type(self.instance_type)}."
-                )
+                if self.instance_groups is not None:
```

Comment:
Are there any existing tests in this package which exercise this block of code? If not, has this flow been manually validated?

Reply:
so there is this check earlier in the method 
`        if "instance_groups" in distribution:
            distribution_config["sagemaker_distribution_instance_groups"] = distribution[
                "instance_groups"
            ]
`
but I just wanted to play it safe since instance_groups is not guarenteed to be a non-NULL value

Reply:
From the docs, it appears that the only case when `instance_type` is not required is when `instance_groups` is passed, so any errors involving both not being set should be caught upstream. Whenever this `else` is entered, this `if` should always be entered, unless `instance_type` passed is somehow not a parameter string or string.

Reply:
I took a look at the docs as well and as @viclzhu mentioned it seems like this check is unnecessary. Just updated the PR to reflect this.